xarray_sql/df_test.py

```diff
@@ -1,387 +1,205 @@
+"""DataFrame functionality tests for xarray-sql."""
+
 import itertools
 import tracemalloc
-import unittest
 
 import numpy as np
 import pandas as pd
+import pytest
 import pyarrow as pa
 import xarray as xr
 
 from .df import explode, read_xarray, block_slices, from_map, pivot, from_map_batched
 
 
+# -------------------------
```

Comment:
(optional): I don't think we need these section comments, but I defer to your judgement if we should keep them (I do see some of their benefit).